pkg/virt-manifest/mapper.go

```diff
@@ -42,52 +43,40 @@ type savedDisk struct {
 	disk v1.Disk
 }
 
-func ExtractPvc(dom *v1.DomainSpec) (*v1.DomainSpec, []savedDisk) {
-	specCopy := &v1.DomainSpec{}
-	model.Copy(specCopy, dom)
-
+func ExtractPvc(dom *v1.DomainSpec) []savedDisk {
 	pvcDisks := []savedDisk{}
-	allDisks := []v1.Disk{}
 
-	for idx, disk := range specCopy.Devices.Disks {
+	for idx, disk := range dom.Devices.Disks {
 		if disk.Type == Type_PersistentVolumeClaim {
 			// Save the disk so we can fix it later
 			diskCopy := v1.Disk{}
 			model.Copy(&diskCopy, disk)
 			pvcDisks = append(pvcDisks, savedDisk{disk: diskCopy, idx: idx})
-
-			// Alter the disk record so that libvirt will accept it
-			disk.Type = Type_Network
-			disk.Source.Protocol = "iscsi"
 		}
-		allDisks = append(allDisks, disk)
 	}
-	// Replace the Domain's disks with modified records
-	specCopy.Devices.Disks = allDisks
 
-	return specCopy, pvcDisks
+	return pvcDisks
 }
 
 // This is a simplified version of the domain creation portion of SyncVM. This is intended primarily
 // for mapping the VM spec without starting a domain.
-func MapVM(con virtwrap.Connection, vm *v1.VM) (*v1.VM, error) {
+func MapVM(con virtwrap.Connection, vm *v1.VM, k8sClient kubernetes.Interface) (*v1.VM, error) {
```

Comment:
I don't want to design something which is clearly going to be broken in the future, so I'm addressing it now, not waiting for us to hit the problem - the latter just pointlessly creates technical debt for us to fix in the future. Even for disks, we're going to hit this in the near future, as for example, the choice of caching mode is dependant on the type of disk backend used in QEMU. There will be plenty of other attributes in the XML we need to fill in as defaults which will also have dependancy on the particular backend config choices made after resolving k8s objects.